### PR TITLE
Add A/A test

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

This change adds an A/A test to make sure we have the Firebase A/B testing setup correctly.

Fixes https://github.com/Automattic/pocket-casts-android/issues/2609

## Testing Instructions
1. Apply the following patch [test.patch](https://github.com/user-attachments/files/16638238/test.patch)
2. Change the variant to `release`
3. Tap on the Profile tab
4. Tap settings
5. Tap 'Pocket Casts Plus'

✅ Verify in the logs the experiment and variation are output
```
Experiment PaywallAATest, variation Control
```

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          4 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>UpgradeLayout</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>Experiment</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>ExperimentProvider</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>Variation</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/analytics/experiments/ExperimentProviderTest.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package au.com.shiftyjelly.pocketcasts.analytics.experiments
+
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+
+class ExperimentProviderTest {
```
Comment (MiSikora) [inline app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/analytics/experiments/ExperimentProviderTest.kt:10]: Can you add tests cases for `null` and unknown variation names?

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/TrackerModule.kt [hunk 2]
```diff
@@ -35,4 +37,10 @@ object TrackerModule {
     internal fun provideFirebaseAnalytics(@ApplicationContext context: Context): FirebaseAnalyticsWrapper {
         return FirebaseAnalyticsWrapper(FirebaseAnalytics.getInstance(context))
     }
+
+    @Provides
+    @Singleton
+    fun provideExperimentProvider(firebaseRemoteConfig: FirebaseRemoteConfig): ExperimentProvider {
+        return ExperimentProvider(firebaseRemoteConfig)
+    }
```
Comment (MiSikora) [inline modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/TrackerModule.kt:45]: You don't need to do this when there is `@Inject` on `ExperimentPorivder`'s constructor.

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/TrackerModule.kt [hunk 3]
```diff
@@ -35,4 +37,10 @@ object TrackerModule {
     internal fun provideFirebaseAnalytics(@ApplicationContext context: Context): FirebaseAnalyticsWrapper {
         return FirebaseAnalyticsWrapper(FirebaseAnalytics.getInstance(context))
     }
+
+    @Provides
+    @Singleton
```
Comment (MiSikora) [inline modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/TrackerModule.kt:42]: `@Singleton` annotation should be used only with objects that are stateful and should have their state kept while the process is live. Synchronization costs of having a singleton object are greater that instantiation of multiple objects in DI scenarios.

**Review by MiSikora — APPROVED** [review]
I think we should think of a more generic way that handles experiments if we want to do more experimentation and in the future integrate with ExPlat.

Otherwise things might be hard to manage and develop with things split between `Feature`, `Experiment`, and `Variation`.

File: app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/analytics/experiments/ExperimentProviderTest.kt (hunk 1 above)
Comment (geekygecko) [reply app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/analytics/experiments/ExperimentProviderTest.kt:10]: Good idea, I have made this change.

File: modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/TrackerModule.kt (hunk 2 above)
Comment (geekygecko) [reply modules/services/analytics/src/main/java/au/com/shiftyjelly/pocketcasts/analytics/di/TrackerModule.kt]: Thanks for pointing this out, I have fixed this up.

**Comment by geekygecko** [comment]
> But I think we should think of a more generic way that handles experiments if we want to do more experimentation and in the future integrate with ExPlat.

For this initial version, I based it on ExPlat and the iOS implementation. I will merge this PR and chat with you about ways of improving it in a future PR.
